src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -465,12 +471,35 @@ int IDebugPendingBreakpoint2.Enable(int fEnable)
                 PendingBreakpoint bp = _bp;
                 if (bp != null)
                 {
+                    Task enableTask = null;
                     _engine.DebuggedProcess.WorkerThread.RunOperation(() =>
                     {
-                        _engine.DebuggedProcess.AddInternalBreakAction(
+                        enableTask = _engine.DebuggedProcess.AddInternalBreakAction(
                             () => bp.EnableAsync(_enabled, _engine.DebuggedProcess)
                         );
                     });
+                    try
+                    {
+                        // Using similar timeout logic to what's used in Bind so we don't block too long
+                        enableTask.Wait(_engine.GetBPLongEnableTimeout());
+                        if (!enableTask.IsCompleted)
```

Comment:
>if (!enableTask.IsCompleted) [](start = 24, length = 28)

Are you running into this being very slow? Being async is problematic for this API because there aren't events that an engine can send to update the UI's breakpoint enable state. So the UI will show one state even though the underlying breakpoint may be in a different state. Not the end of the world, but not ideal either.